commit 80a23d025f39082e42e01fbc89b33e9add5fa811 (refs/changes/25/2225/2)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2017-07-25T22:27:53+00:00 (2 years, 2 months ago)
    
    DOCKER-1070 Don't recreate "sdc_docker" rule when it's been disabled

diff --git a/lib/backends/sdc/containers.js b/lib/backends/sdc/containers.js
index a99acac..516d192 100644
--- a/lib/backends/sdc/containers.js
+++ b/lib/backends/sdc/containers.js
@@ -265,10 +265,6 @@ function addExposeFirewallRules(opts, payload, callback) {
                 continue;
             }
 
-            if (!rules[r].enabled) {
-                continue;
-            }
-
             if (parsed.action !== 'allow') {
                 continue;
             }
@@ -286,14 +282,20 @@ function addExposeFirewallRules(opts, payload, callback) {
                 tcpRuleFound = true;
                 opts.log.debug({ rule: rules[r] },
                     'TCP docker expose rule found: not adding');
-                addRulesToPayload(payload, rules[r]);
+
+                if (parsed.enabled) {
+                    addRulesToPayload(payload, rules[r]);
+                }
             }
 
             if (parsed.protocol === 'udp' && !udpRuleFound) {
                 udpRuleFound = true;
                 opts.log.debug({ rule: rules[r] },
                     'UDP docker expose rule found: not adding');
-                addRulesToPayload(payload, rules[r]);
+
+                if (parsed.enabled) {
+                    addRulesToPayload(payload, rules[r]);
+                }
             }
         }
 
